### PR TITLE
JCLOUDS-1492 Dimension Data Feature API Predicates are not usable

### PR DESCRIPTION
**Pull request by trevorflanagan**
Fix for issue identified in https://issues.apache.org/jira/browse/JCLOUDS-1492

### CODE REVIEW

File: dimensiondata/src/main/java/org/jclouds/dimensiondata/cloudcontrol/domain/OsImage.java [hunk 1]
```diff
@@ -31,6 +32,7 @@
       type = TYPE;
    }
 
+   @Nullable
```
Comment (trevorflanagan) [inline dimensiondata/src/main/java/org/jclouds/dimensiondata/cloudcontrol/domain/OsImage.java:35]: unrelated to fix, but this is breaking our live tests

File: dimensiondata/src/test/java/org/jclouds/dimensiondata/cloudcontrol/features/NetworkApiLiveTest.java [hunk 2]
```diff
@@ -42,114 +42,134 @@
...
-
-   private FirewallRule findById(List<FirewallRule> collection, String id) {
-      for (FirewallRule rule : collection) {
-         if (rule.id().equals(id)) {
-            return rule;
-         }
-      }
-      return null;
-   }
+public class NetworkApiLiveTest extends BaseDimensionDataCloudControlApiLiveTest
```
Comment (nacx) [inline dimensiondata/src/test/java/org/jclouds/dimensiondata/cloudcontrol/features/NetworkApiLiveTest.java]: Can you please reformat this file according to the jclouds style guidelines?

Comment (trevorflanagan) [reply dimensiondata/src/test/java/org/jclouds/dimensiondata/cloudcontrol/features/NetworkApiLiveTest.java]: Thanks for looking @nacx , I put a new image on my machine recently so had to set everything up again. Forgot to apply the formatting settings.

**Comment by nacx** [comment]
Cherry-picked to 2.1.x too. Thanks!

**Pull request merged**